src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -184,6 +200,12 @@ protected override GetHeadersPayload BuildGetHeadersPayload()
         {
             INetworkPeer peer = this.AttachedPeer;
 
+            if (this.isGateway && !peer.IsWhitelisted())
+            {
+                this.logger.LogTrace("Node is a gateway, cannot sync from non whitelisted peer.");
+                return null;
+            }
+
```

Comment:
later in the code if we are the gateway and peer is whitelisted and it is a C# node we will ask for PH. 
We should ask for normal headers.

Reply:
yes, actually we don't support C# whitelisted node as gateway. If we need to, we can ask for legacy peer but we should implement even a way to communicate using legacy node when the non-gateway c# node has a block to announce to us

Reply:
why not just do the inverse?

```
if (this.isGateway && peer.IsWhitelisted())
  return base.BuildGetHeadersPayload();
```
And bellow remove any whitelisting reference